src/fstab-generator/fstab-generator.c

```diff
@@ -744,11 +749,22 @@ static int parse_fstab(bool initrd) {
                         flags |= rw_only * MOUNT_RW_ONLY |
                                  automount * MOUNT_AUTOMOUNT;
 
+                        is_sysroot = path_equal(where, "/sysroot");
+                        is_sysroot_usr = path_equal(where, "/sysroot/usr");
+
                         const char *target_unit =
                                 initrd ?               SPECIAL_INITRD_FS_TARGET :
+                                is_sysroot ?           SPECIAL_INITRD_ROOT_FS_TARGET :
+                                is_sysroot_usr ?       SPECIAL_INITRD_USR_FS_TARGET :
```

Comment:
Ah, so it does! I looked less closely at that one (since I was mostly interested in just root fs). I'll make an update that can equivalently cover that case as well sometime in the next day or so

Reply:
Should be fixed now